enhancements/insights/on-demand-data-gathering.md

```diff
@@ -0,0 +1,340 @@
+- wait for the job completion and if the processing part was successful then download the last Insights analysis report
+- check that the Insights analysis report ID is equal to the gathering job request ID. If the IDs are equal then update the corresponding job status with `AnalysisAvailable=True` condition and continue with the following step below, otherwise update the corresponding job status with `AnalysisAvailable=False` condition and do nothing 
+- based on the status of the last `datagather.insights.openshift.io` CR update the `insightsoperator.operator.openshift.io` CR and also the clusteroperator conditions
+- prune the old job definitions and `datagather.insights.openshift.io` CRs
+  
+As mentioned above the operator will work with CRs created only in the `openshift-insights` namespace and will create jobs only in this namespace. The periodically created jobs will have the `periodic` and current timestamp prefix. To start on-demand data gathering, the user must create a new CR with a unique name.
+
+### Risks and Mitigations
+
+Risk: Inconsistent Insights analysis results. Assume following situation:
+- periodic gathering is running in the Insights operator (let's call it gather A)
+- user runs a new on demand data gathering (let's call it gather B) e.g 30 seconds after gather A
+- gather A and B finish at essentially the same time
+- the external processing service (in the console.redhat.com) provides only one recent Insights analysis result, and there is no guarantee which (of the two) it is.
+- this can be very confusing, because user running gather B can get result of the gather A (e.g there could be different gatherers enabled, which would produce different results)
+
+Mitigation: The gathering job will only check whether the corresponding archive was successfully processed or not. This requires a new API endpoint to be implemented in the CCX processing data pipeline ([API documentation here](https://console.redhat.com/docs/api/insights-results-aggregator/v2)). This request is tracked in https://issues.redhat.com/browse/CCXDEV-9980. 
+The original requirement for the external processing was to provide the most recent Insights analysis and recommendations in the HAC (Hybrid Application Console) and ACM (Advanced Cluster Manager) and there is no use case for maintaining a history of Insights recommmedations/analyses in the cluster, plus it would take considerable effort to enable such a history. 
+
+Risk: Problematic access to archives of gathered data (e.g for inspecting the data). The archives are currently stored in a container. This is problematic because once the Pod/job is finished, the container can no longer be accessed. 
+
+Mitigation: Introduce a new config option for the Insights operator. This option allows users to specify a volume name and path that can then be  mounted to a job, and the job will also save the archives to the mounted volume. 
```

Comment:
Nice, so this would allow me for example, to create a persistent volume and attach it to all jobs being created? Do you see this as a global option or a per-job option?

Reply:
Yes and yes I see it as a global option.

Reply:
I have expanded the wording here too.